### PR TITLE
Fix/impruvment fixes

### PR DESCRIPTION
**Pull request by VSemikozInostudio**
Performance improvement, logout fixes

### CODE REVIEW

File: lib/src/room/rooms.dart [hunk 1]
```diff
@@ -75,33 +76,52 @@ class Rooms extends DelegatingIterable<Room> implements Contextual<Rooms> {
...
-          }
-
-          return room.merge(otherRoom!);
-        }),
-        // All other rooms that were not merged, they're new
-        ...other.where(
-          (otherRoom) => !any((room) => otherRoom.equals(room)),
-        ),
-      ], growable: false),
+    final currList = toList();
```
Comment (Deltenman) [inline lib/src/room/rooms.dart:79]: Is that legal wrote just "toList()"?

Comment (VSemikozInostudio) [reply lib/src/room/rooms.dart:79]: mean this.toList(). Short version because dart analysis.

Comment (VSemikozInostudio) [reply lib/src/room/rooms.dart:79]: toList is an internal method for Rooms Model

Comment (Deltenman) [reply lib/src/room/rooms.dart:79]: Understood, thanks

**Review by Deltenman — APPROVED** [review]

**Review by esleptsovaINO — APPROVED** [review]

**Pull request merged**